reader.go

```diff
@@ -67,34 +68,40 @@ func (sfr *FactsetReader) download(filePath string, fileName string, dest string
 	return nil
 }
 
-func (sfr *FactsetReader) getLastVersion(files []os.FileInfo, searchedRes string) (string, error) {
-	recFile := &struct {
-		name string
-		vers int
+func (sfr *FactsetReader) getLastVersion(files []os.FileInfo, searchedFileName string) (string, error) {
+	foundFile := &struct {
+		name         string
+		majorVersion int
+		minorVersion int
 	}{}
 
-	r := regexp.MustCompile("[0-9]+\\.zip$")
 	for _, file := range files {
 		name := file.Name()
-		if !strings.Contains(name, searchedRes) {
+		if !strings.Contains(name, searchedFileName) {
 			continue
 		}
-		s := r.FindStringSubmatch(name)[0]
 
-		v, err := strconv.Atoi(strings.TrimSuffix(s, ".zip"))
+		fullVersion, err := GetFullVersion(name)
```

Comment:
It has to be in the interface, so yes, it has to be public, but not static (stateless), so I changed it to be part of interface.